src/Tmds.Ssh/SftpChannel.cs

```diff
@@ -1079,10 +1082,11 @@ private async ValueTask DownloadFileAsync(string remotePath, string localPath, l
             permissions = attributes.Permissions;
         }
 
-        using FileStream localFile = OpenFileStream(localPath, overwrite ? FileMode.Create : FileMode.CreateNew, FileAccess.Write, FileShare.None, permissions!.Value);
+        using FileStream? localFile = localPath is null ? null : OpenFileStream(localPath, overwrite ? FileMode.Create : FileMode.CreateNew, FileAccess.Write, FileShare.None, permissions.Value);
+        destination ??= localFile;
```

Comment:
Add a `Debug.Assert(destination is not null)` to eliminate the nullability warnings.

Reply:
done